modelforge/potential/models.py

```diff
@@ -540,19 +540,41 @@ def apply_bwd(res, grads):
         return apply, model_params, model_buffer
 
 
+from modelforge.potential.parameters import (
+    ANI2xParameters,
+    PhysNetParameters,
+    SchNetParameters,
+    PaiNNParameters,
+    SAKEParameters,
+    TensorNetParameters,
+)
+from modelforge.train.parameters import TrainingParameters, RuntimeParameters
+from modelforge.dataset.dataset import DatasetParameters
+
+
 class NeuralNetworkPotentialFactory:
     """
     Factory class for creating instances of neural network potentials for training/inference.
     """
 
     @staticmethod
-    def generate_model(
+    def generate_potential(
         *,
         use: Literal["training", "inference"],
-        model_parameter: Dict[str, Union[str, Any]],
+        potential_parameter: Union[
+            ANI2xParameters,
+            SAKEParameters,
+            SchNetParameters,
+            PhysNetParameters,
+            PaiNNParameters,
+            TensorNetParameters,
+        ],
         simulation_environment: Literal["PyTorch", "JAX"] = "PyTorch",
```

Comment:
Agreed, we can remove the defaults here and control it via the runtime config

I can access this attribute in the initialized Pydantic model and set it to a different value at runtime (only necessary for testing), right?

Reply:
I initially removed it, then realized we use this in tests.  I changed the code so that it would use the environment from the runtime_parameters (if provided), which it wasn't doing.  If we set simulation_environment explicitly, it will override it.  If not provided in either case, it defaults to PyTorch.  Again, I think mostly just useful for testing.